### PR TITLE
logic was inverted for passing loss_kwargs to forward pass

### PR DESCRIPTION
**Pull request by winglian**
# What does this PR do?
Bug introduced in https://github.com/huggingface/transformers/pull/34915 inverted the loss kwargs logic.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -1180,7 +1183,7 @@ def forward(
 
         loss = None
         if labels is not None:
-            loss = self.loss_function(logits=logits, labels=labels, vocab_size=self.config.vocab_size, **kwargs)
+            loss = self.loss_function(logits=logits, labels=labels, vocab_size=self.config.vocab_size, num_items_in_batch=num_items_in_batch, **kwargs)
```
Comment (ArthurZucker) [inline src/transformers/models/llama/modeling_llama.py:1186]: no, the whole point of having kwargs here is not not introduce unwanted training specific args 😓

**Review by ArthurZucker — COMMENTED** [review]
Linked to #35113, will do a patch for this

File: src/transformers/models/llama/modeling_llama.py (hunk 1 above)
Comment (winglian) [reply src/transformers/models/llama/modeling_llama.py:1186]: right, the problem is I have to pop it from kwargs to keep from passing it to `model(...)` above
